libcylinder/src/key/load.rs

```diff
@@ -202,34 +255,49 @@ mod tests {
         writeln!(temp_file, "{}", private_key.as_hex())
             .expect("Unable to write private key to file");
 
-        let retrieved_private_key =
-            load_user_key(None, default_path).expect("Unable retrieve key from file");
+        let retrieved_private_key = load_key(&current_user_key_name(), &current_user_search_path())
+            .expect("Unable retrieve key from file");
+
+        assert_eq!(
+            retrieved_private_key.unwrap().into_bytes(),
+            private_key.into_bytes()
+        );
 
-        assert_eq!(retrieved_private_key.into_bytes(), private_key.into_bytes(),);
+        env::remove_var("CYLINDER_KEY_NAME");
+        env::remove_var("CYLINDER_PATH");
+        assert!(env::var("CYLINDER_KEY_NAME").is_err());
+        assert!(env::var("CYLINDER_PATH").is_err());
     }
 
-    // tests that when an environment variable is given as the default path load_user_key
-    // successfully returns the private key
+    /// Tests that when the CYLINDER_PATH environment variable is set with multiple paths,
+    /// `key_load` will successfully retrieve the private key if passed
+    /// current_user_key_name() and current_user_search_path() as arguments.
+    ///
+    /// 1. Create a private key file in a temporary directory and write a key to the file.
+    /// 2. Set the CYLINDER_KEY_NAME env variable to the name of the private key file.
+    /// 3. Set the CYLINDER_PATH env variable to be two different file paths, one of which
```

Comment:
Is the correct path the first or second?  Interesting to verify that the fail-over works.

Reply:
Updated